src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/FragmentsValidationInterpreter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.SchemaDriven.Processors
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.DocAsCode.Common;
+
+    public class FragmentsValidationInterpreter : IInterpreter
+    {
+        public bool CanInterpret(BaseSchema schema)
+        {
+            return schema == null || schema.ContentType != ContentType.Uid;
+        }
+
+        public object Interpret(BaseSchema schema, object value, IProcessContext context, string path)
+        {
+            if (value is IDictionary<string, Object> || value is IDictionary<object, Object> || value is IList<object>)
+            {
+                return value;
+            }
+
+            if (schema?.MergeType == MergeType.Key)
+            {
+                return value;
+            }
+
+            if (schema?.Tags != null && schema.IsLegalInFragments())
+            {
+                return value;
+            }
+
+            if (value is string && schema == null)
```

Comment:
>schema == null [](start = 35, length = 14)

What about check first? then no need to `schema?.`

Reply:
this is a `and` expression, we still need to check shema after this expression.

---
In reply to: [176642487](https://github.com/dotnet/docfx/pull/2546#discussion_r176642487) [](ancestors = 176642487)